src/WebJobs.Extensions.DurableTask/Options/DurableTaskOptions.cs

```diff
@@ -237,15 +288,20 @@ internal void Validate()
 
         internal bool IsDefaultHubName()
         {
-            return string.Equals(this.defaultHubName, this.hubName, StringComparison.OrdinalIgnoreCase);
+            return string.Equals(this.defaultHubName, this.HubName, StringComparison.OrdinalIgnoreCase);
+        }
+
+        private static bool IsInDevelopment()
+        {
+            return Environment.GetEnvironmentVariable("WEBSITE_SLOT_NAME") == null;
         }
 
         private static bool IsInNonProductionSlot()
         {
             var slotName = Environment.GetEnvironmentVariable("WEBSITE_SLOT_NAME");
 
             // slotName can be null in a test environment
-            if (slotName != null && !string.Equals(slotName, "Production", StringComparison.OrdinalIgnoreCase))
+            if (slotName == null || !string.Equals(slotName, "Production", StringComparison.OrdinalIgnoreCase))
```

Comment:
I get the desire to split out the case where it is a development environment, but I'm not certain that this implementation made it any clearer, as this method now returns `true` if the environment is dev environment, and the method name no longer really makes sense, as development environments are not in non-production slots.